src/main/java/org/jabref/logic/integrity/UnicodeNormalFormCCheck.java

```diff
@@ -0,0 +1,29 @@
+package org.jabref.logic.integrity;
+
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+
+/**
+ * Detect any Unicode characters that is not in NFC format
+ */
+public class UnicodeNormalFormCCheck implements EntryChecker {
+
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        List<IntegrityMessage> results = new ArrayList<>();
+        for (Map.Entry<Field, String> field : entry.getFieldMap().entrySet()) {
+            String normalizedString = Normalizer.normalize(field.getValue(), Normalizer.Form.NFC);
```

Comment:
I checked the JavaDoc of normalizer now. It is available at https://docs.oracle.com/javase/6/docs/api/java/text/Normalizer.html.

Please use [isNormalized](https://docs.oracle.com/javase/6/docs/api/java/text/Normalizer.html#isNormalized(java.lang.CharSequence,%20java.text.Normalizer.Form)). That should be faster.